nginx-php-moodle/bin/watch_certificate_dir.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/with-contenv bash
+#
+# Copyright (c) 2018 SD Elements Inc.
+#
+#  All Rights Reserved.
+#
+# NOTICE:  All information contained herein is, and remains
+# the property of SD Elements Incorporated and its suppliers,
+# if any.  The intellectual and technical concepts contained
+# herein are proprietary to SD Elements Incorporated
+# and its suppliers and may be covered by U.S., Canadian and other Patents,
+# patents in process, and are protected by trade secret or copyright law.
+# Dissemination of this information or reproduction of this material
+# is strictly forbidden unless prior written permission is obtained
+# from SD Elements Inc..
+# Version
+
+# Source shtdlib
+# shellcheck disable=SC1091
+source /usr/local/bin/shtdlib.sh
+
+echo "Watching ${CERT_DIR} for updated certs..."
+
+function reload_nginx {
+    echo "Detected files modified in certificate directory..."
+    echo "Linking certificate files..."
+    ln -sf "${CERT_DIR}/${CERT_DOMAIN}.fullchain.pem" /etc/nginx/ssl/moodle.crt
```

Comment:
The file name never changes.. so why do we need to keep re-linking?

Reply:
We have to create the initial link but after that the file does stay the same. I'm not sure if creating the link every time after will have any unforeseen issues though. I just kept this behaviour so the script stayed pretty simple.

Reply:
I think its better to make the script idempotent - check if a link exists and act accordingly. Make it if it doesn't exist and log a message if it does.

Reply:
Good point, I've added a check to see if the file is a symlink.